src/app/core/public/api/impl/Selections.ts

```diff
@@ -22,21 +23,39 @@ export class SelectionsImpl implements Selections {
         return this.state.selectionsManager.length();
     }
     public get isEmpty(): boolean {
-        return (this.length > 0);
+        return (this.length === 0);
     }
 
+    public get all(): Obj[] {
+        return this.selections.map((id) => this.state.getObj(id)!);
+    }
     public get components(): Component[] {
         return this.selections.filter((id) => (this.doc.hasComp(id)))
             .map((id) => this.state.constructComponent(id));
     }
+    public get wires(): Wire[] {
+        return this.selections.filter((id) => (this.doc.hasWire(id)))
+            .map((id) => this.state.constructWire(id));
+    }
```

Comment:
I think this would look better with the `.filter` on its own line like the `forEach` function below